crates/ruff_server/src/session/index/ruff_settings.rs

```diff
@@ -88,7 +88,11 @@ impl RuffSettingsIndex {
             .filter(|entry| entry.file_type().is_dir())
             .map(DirEntry::into_path)
         {
-            if let Some(pyproject) = settings_toml(&directory).ok().flatten() {
+            if let Some(pyproject) = find_settings_toml(&directory).ok().flatten() {
```

Comment:
I think we should only do this once, for the root path.

Reply:
There's no need to search in parent directories for _every_ directory, since there's at most one pyproject above the root path.